server.js

```diff
@@ -765,12 +766,44 @@ cache(function(data, match, sendBadge, request) {
     }
     try {
       var data = JSON.parse(buffer);
-      // Grab the latest stable version, or an unstable
+
       var versions = Object.keys(data.package.versions);
-      var version = latestVersion(versions);
-      var vdata = versionColor(version);
-      badgeData.text[1] = vdata.version;
-      badgeData.colorscheme = vdata.color;
+      var stableVersion = latestVersion(versions);
+      var vdata = versionColor(stableVersion);
+
+      var badgeText = null;
+      var badgeColor = null;
+
+      switch (info) {
+          case 'v':
+              badgeText = vdata.version;
+              badgeColor = vdata.color;
+              break;
+          case 'vpre':
+              if (versions[0] !== undefined) {
+
+                  var unstableVersion = null;
+                  for (var i = 0; i < versions.length; i++) {
+                      var version = versions[i];
+
+                      if (version.charAt(0) !== 'v') {
```

Comment:
If there are no versions starting without prefix v, it's all considered a stable release.

Hmm?

	var version = "v1.2.3";
	
	version[0]; // "v"
	version[1]; // "1"
	version.charAt(0); // "v"
	version.charAt(1); // "1"

Or am I wrong?